lib/pure/streams.nim

```diff
@@ -243,6 +243,9 @@ proc readDataStr*(s: Stream, buffer: var string, slice: Slice[int]): int =
     result = s.readDataStrImpl(s, buffer, slice)
   else:
     # fallback
+    when declared(prepareMutation):
```

Comment:
I didn't add a nimvm check here because this code branch isn't supported by the VM anyway